src/coreclr/vm/method.hpp

```diff
@@ -64,8 +64,8 @@ enum class AsyncMethodKind
 {
```

Comment:
Would it be better to make this a bit field?

- ReturnsTaskOrValueTask = 0x1, // The method returns Task or ValueTask
- Async = 0x2, // The method IL in metadata has MethodImplAttributes.Async 
- AsyncCall = 0x4, // The entrypoint of this method has async calling convention

I took the bit names from the managed type system. I am not attached to these names. If we come up with different names, we should update the managed type system to match.

Reply:
> Would it be better to make this a bit field?

Possibly. While it may be difficult to name combinations of traits, each individual trait seems easier to name.

Reply:
> each individual trait seems easier to name.

Yep, individual traits are easier to name and explain.

> MethodImplAsync if we go for more verbose names.

More verbose self-describing names are good to avoid confusion. `MethodImplAsync` flag / `IsMethodImplAsync` would be fine with me.

Reply:
> MethodImplAsync

The managed type system has similar prior art: The property for `MethodImplAttributes.Runtime` is called `MethodDesc.IsRuntimeImplemented` since `MethodDesc.IsRuntime` would look odd.